// test_react/dist/components/Test/Test.js
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
define(["require", "exports", "react"], function (require, exports, react_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.Test = void 0;
    react_1 = __importDefault(react_1);
    class Test extends react_1.default.Component {
        constructor() {
            super(...arguments);
            this.state = {
                inp: this.props.label,
                error: this.props.error,
            };
            this.onChange = (event) => {
                this.setState(() => ({
                    inp: event.target.value
                }));
            };
            this.triggerError = () => {
                this.setState(() => ({
                    error: true,
                }));
                this.props.errorTriggered(true);
            };
            this.classes = () => {
                return `test ${this.state.error ? 'test--error' : ''}`;
            };
        }
        render() {
            return (react_1.default.createElement("div", { className: this.classes() },
                react_1.default.createElement("h2", null, "Test component to test props and events"),
                react_1.default.createElement("span", null,
                    " Prop label: ",
                    react_1.default.createElement("strong", null, this.state.inp)),
                react_1.default.createElement("br", null),
                react_1.default.createElement("br", null),
                react_1.default.createElement("br", null),
                react_1.default.createElement("input", { type: "text", value: this.state.inp, onChange: this.onChange }),
                react_1.default.createElement("br", null),
                react_1.default.createElement("br", null),
                react_1.default.createElement("br", null),
                react_1.default.createElement("button", { onClick: this.triggerError }, "Click to trigger error"),
                this.state.error &&
                    react_1.default.createElement("div", null,
                        react_1.default.createElement("p", null, "Only visible if error triggered"))));
        }
    }
    exports.Test = Test;
});
